newlib/libc/tinystdio/vfprintf.c

```diff
@@ -983,8 +983,11 @@ int vfprintf (FILE * stream, const CHAR *fmt, va_list ap_orig)
 
             pnt = NULL;
 #endif
-
-            if (c == 'c') {
+            if (c == 'n') {
```

Comment:
This needs to be gated by a global configuration parameter so that we can remove the code for targets which prefer the safety of not allowing `%n` while still enabling it for targets requiring strict standards conformance.